### PR TITLE
hideOnLeave option for context menu

### PR DESCRIPTION
**Pull request by kamranjon**
add hideOnLeave boolean for context menu and expose onMouseLeave for extension if needed.

### CODE REVIEW

**Comment by vkbansal** [comment]
Thanks for your contribution. Can you please explain the use case for this change?

**Comment by kamranjon** [comment]
This allows the menu to be hidden when the users mouse leaves the menu/submenu area - it is useful just as an interface feature. This also affords the user to hook into the onMouseLeave event so they can do whatever they want.

File: src/ContextMenu.js [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
-import { cssClasses, callIfExists } from './helpers';
+import { cssClasses, callIfExists, store } from './helpers';
 
 export default class ContextMenu extends Component {
     static propTypes = {
         id: PropTypes.string.isRequired,
         className: PropTypes.string,
+        hideOnLeave: PropTypes.bool,
         onHide: PropTypes.func,
+        onMouseLeave: PropTypes.func,
```
Comment (vkbansal) [inline src/ContextMenu.js]: Document `onMouseLeave` in README

Comment (kamranjon) [inline src/ContextMenu.js]: Updated the readme to list onMouseLeave as contextmenu property

**Pull request merged**